File: app/src/main/java/cm/aptoide/pt/billing/transaction/Transaction.java
package cm.aptoide.pt.billing.transaction;

public interface Transaction {
  String getCustomerId();

  String getProductId();

  String getId();

  boolean isNew();

  boolean isCompleted();

  boolean isPendingAuthorization();

  boolean isProcessing();

  boolean isFailed();

  enum Status {
    NEW, PENDING_SERVICE_AUTHORIZATION, PROCESSING, COMPLETED, FAILED
  }
}

File: payments/base/payment-manager/src/main/java/cm/aptoide/pt/payment_manager/transaction/Transaction.kt
package cm.aptoide.pt.payment_manager.transaction

import androidx.annotation.Keep
import kotlinx.coroutines.flow.Flow

interface Transaction {
  val status: Flow<TransactionStatus>
}

@Keep
enum class TransactionStatus {
  PENDING,
  PENDING_SERVICE_AUTHORIZATION,
  SETTLED,
  PROCESSING,
  COMPLETED,
  PENDING_USER_PAYMENT,
  INVALID_TRANSACTION,
  FAILED,
  CANCELED,
  DUPLICATED,
  CHARGEBACK,
  REFUNDED,
  FRAUD,
  PENDING_VALIDATION,
  PENDING_CODE,
  VERIFIED,
  EXPIRED
}
